### PR TITLE
feat: [CDS-68853]: add commitIds of rettaform resources to step output

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49910)

### CODE REVIEW

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger feature-build

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger feature-build

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStep.java [hunk 1]
```diff
@@ -355,6 +362,33 @@ public StepResponse handleTaskResultWithSecurityContext(Ambiance ambiance,
...
+
+  @NotNull
+  private Map<String, String> getGitRevisionsOutput(
+      Ambiance ambiance, TerraformDestroyStepParameters parameters, TerraformTaskNGResponse terraformTaskNGResponse) {
+    Map<String, String> outputKeys = new HashMap<>();
+    if (parameters.getConfiguration().getType().getDisplayName().equalsIgnoreCase(
+            TerraformStepConfigurationType.INLINE.getDisplayName())) {
+      outputKeys = helper.getRevisionsMap(parameters.getConfiguration().getSpec().getVarFiles(),
+          terraformTaskNGResponse.getCommitIdForConfigFilesMap());
+    } else if (parameters.getConfiguration().getType().getDisplayName().equalsIgnoreCase(
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStep.java:382]: Do we need this else if statement when it's INHERIT_FROM_PLAN?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStepV2.java [hunk 2]
```diff
@@ -385,7 +400,7 @@ private TerraformTaskNGParametersBuilder getTerraformTaskNGParametersBuilderInhe
     TerraformConfig terraformConfig = helper.getLastSuccessfulApplyConfig(stepParameters, ambiance);
 
     builder.workspace(terraformConfig.getWorkspace())
-        .varFileInfos(helper.prepareTerraformVarFileInfo(terraformConfig.getVarFileConfigs(), ambiance))
+        .varFileInfos(helper.prepareTerraformVarFileInfo(terraformConfig.getVarFileConfigs(), ambiance, true))
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStepV2.java:403]: was it checked when destroy inherits from apply? will we have the original Identifier?

Comment (tmacari-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStepV2.java:403]: Yes

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStep.java (hunk 1 above)
Comment (tmacari-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformDestroyStep.java:382]: Yes, in that case we wont display commit ids for var files

**Comment by tmacari-harness** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStep.java [hunk 3]
```diff
@@ -294,10 +293,18 @@ public StepResponse handleTaskResultWithSecurityContext(Ambiance ambiance,
...
 
-      stepResponseBuilder.stepOutcome(StepResponse.StepOutcome.builder()
-                                          .name(TerraformPlanOutcome.OUTCOME_NAME)
-                                          .outcome(tfPlanOutcomeBuilder.build())
-                                          .build());
+      Map<String, String> outputKeys = helper.getRevisionsMap(
+          planStepParameters.getConfiguration().getVarFiles(), terraformTaskNGResponse.getCommitIdForConfigFilesMap());
+
+      stepResponseBuilder
+          .stepOutcome(StepResponse.StepOutcome.builder()
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStep.java:300]: Duplicate in builder. Please have only one stepOutcome

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by vlica-harness — APPROVED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStep.java (hunk 3 above)
Comment (tmacari-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terraform/TerraformPlanStep.java:300]: it supports multiple:
"@Wither @Singular Collection<StepOutcome> stepOutcomes;"

**Review by puthrayaharness — APPROVED** [review]

**Comment by tmacari-harness** [comment]
trigger review

**Comment by tmacari-harness** [comment]
trigger review

**Pull request merged**